internal/util/typeutil/schema.go

```diff
@@ -275,20 +275,24 @@ func AppendFieldData(dst []*schemapb.FieldData, src []*schemapb.FieldData, idx i
 			switch srcVector := fieldType.Vectors.Data.(type) {
 			case *schemapb.VectorField_BinaryVector:
 				if dstVector.GetBinaryVector() == nil {
+					srcToCopy := srcVector.BinaryVector[idx*(dim/8) : (idx+1)*(dim/8)]
 					dstVector.Data = &schemapb.VectorField_BinaryVector{
-						BinaryVector: srcVector.BinaryVector[idx*(dim/8) : (idx+1)*(dim/8)],
+						BinaryVector: nil,
 					}
+					copy(dstVector.Data.(*schemapb.VectorField_BinaryVector).BinaryVector, srcToCopy)
```

Comment:
`BinaryVector` is not pre-allocated, so copy will do nothing.  Maybe it shall be like this
```
BinaryVector: make([]byte, len(srcToCopy),
```

Reply:
OK, got it. thx for pointing out this.